python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -739,6 +769,19 @@ def resolve_inner(self, asset_graph: AssetGraph) -> AbstractSet[AssetKey]:
     def to_serializable_asset_selection(self, asset_graph: AssetGraph) -> "AssetSelection":
         return self._replace(child=self.child.to_serializable_asset_selection(asset_graph))
 
+    def __str__(self) -> str:
+        if self.depth is None:
+            base = f"*({self.child})"
+        elif self.depth == 0:
+            base = str(self.child)
+        else:
+            base = f"{'+' * self.depth}({self.child})"
+
+        if self.include_self:
+            return base
+        else:
+            return f"{base} - ({self.child})"
```

Comment:
I think this is actually an inaccurate translation -- the `include_self` argument being `False` does not mean that it is impossible for the operand to show up in the result set. For example, imagine you have the graph

`A -> B -> C -> D`, and you make the selection "DownstreamAssetSelection(KeysAssetSelection(B, C, D), include_self=False)"

This would resolve to `{C, D}`, because `C` and `D` are both downstream of an element of the input set. Setting `include_self` to True would change the output to `{B, C, D}`.

But in this translation, this would imply that if `include_self` was `False`, then we'd get the empty set as output.

Coming up with an accurate translation here is pretty tough, because I think the default assumption really is that `include_self` is True (i.e. `"(foo)+"` would be assumed to always include foo). Maybe we just need to have a kinda jank explicit language for this case:

```python
if self.include_self:
    return base
else:
    return f"uninclusive_downstream({self.child})"
```

The true logical expression (assuming we don't have an explicit operator for this) is pretty tough to define.

Reply:
> This would resolve to {C, D}, because C and D are both downstream of an element of the input set. Setting include_self to True would change the output to {B, C, D}.

Looking at the implementation, I think that the result would be the empty set? Maybe this is unexpected though?